tests/scripts/set_psa_test_dependencies.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
+
+"""Edit test cases to use PSA dependencies instead of classic dependencies.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import sys
+
+CLASSIC_DEPENDENCIES = frozenset([
```

Comment:
I don't think this is an oversight but I prefer to double check. MBEDTLS_PK_C, MBEDTLS_PK_PARSE_C, MBEDTLS_PK_WRITE_C and MBEDTLS_MD_C are not in this list and thus always kept in dependencies. Not possible to get rid of them ?

Reply:
I deliberately kept the high-level modules, and only removed the low-level modules, because the low-level modules are the ones that are tied to specific algorithms, and because that's what code using `MBEDTLS_USE_PSA_CRYPTO` will need. I'm not convinced that this was the right choice at this stage though.

Reply:
Unless we find something wrong with the PR as it is, I propose to take what we have now, and if it's suboptimal, fix it in a follow-up.

On a general note, while making these changes, I gradually realized that removing old dependencies and determining the new dependencies were almost entirely independent problems: the fact that we needed it as a conversion was just happenstance. So I expect that once we've finished this translation, we'll think about the problem differently, and we can't expect to get the architecture right the first time.

Reply:
ok, fine by me. We will remove them when we encounter a case where some test cases are not run while they should.